server/controllers/todos.js

```diff
@@ -0,0 +1,73 @@
+const ToDoList = require("../models/ToDoList");
+const ToDo = require("../models/ToDo");
+const {StatusCodes} = require("http-status-codes");
+const {BadRequestError, NotFoundError} = require("../errors");
+
+const getAllTodos = async (req, res) => {
+  const todos = await ToDo.find({owner: req.user.userId}).sort("createdAt");
+  res.status(StatusCodes.OK).json({todos, count: todos.length});
+};
+
+const getTodo = async (req, res) => {
+  const {
+    user: {userId},
+    params: {id: todoId},
+  } = req;
+  const todo = await ToDo.findOne({
+    _id: todoId,
+    owner: userId,
+  });
+  if (!todo) throw new NotFoundError(`No entry found with ID no. ${todoId}`);
```

Comment:
This error message won't be meaningul to users. I don't think including the todoId is necessary.

Reply:
Here you realize the errors are thrown also for the purpose of displaying custom message to UI. What would you suggest the message be?

Reply:
I may be wrong here, but it seems to me that getting one to do would be useful for doing things like updating or deleting that single to do item. Possibly this could be used if a user is searching for a to do item? If the to do is not found, an error message like "No entry found. Create a to do item." would be more meaningful to the user.

Reply:
The style of this error message, which appears also for the `updateTodo` function and the `getList` and `updateList` functions in `/server/controllers/list.js`, is copied from the video tutor's code in the Node.js class. I suspected the tutor had it this way only to serve the developer, and I left it as is here only to be conservative lest there's something I don't know. I'll confirm that there isn't a useful thing served by the id# and edit them to "No todo list/item is found that matches your request."

There are needs to get a single todo or todo list to be displayed, to populate an update form, or to remove it from db. Though I doubt that for a todo item, we don't have a MVP clearly defined so it's hard to say.